feat : Match 기능 구현

Newbiekk-kkh opened the pull request:
## #️⃣연관된 이슈

> #17 

## 📝작업 내용

매칭 요청,
매칭 수락/거절.
받은 매칭 전체보기,
보낸 매칭 전체보기,
매칭 삭제(취소)

### 스크린샷 (선택)

## 💬리뷰 요구사항(선택)

Newbiekk-kkh committed:
feat : 매칭 요청 생성 기능 구현

Newbiekk-kkh committed:
feat : 매칭 수락/거절 기능 구현

Newbiekk-kkh committed:
feat : 매칭 전체 조회 기능 구현

Newbiekk-kkh committed:
feat : 매칭 단일 조회 기능 구현

Newbiekk-kkh committed:
refactor : 매칭 CRUD 리팩토링

매칭 단일 조회, 매칭 전체 조회 - 삭제

받은 매칭 전체 조회, 보낸 매칭 전체 조회 - 추가
보낸 매칭 삭제 - 추가

Newbiekk-kkh committed:
Merge remote-tracking branch 'origin/develop' into feature/match

# Conflicts:
#	src/main/java/com/example/gamemate/global/constant/ErrorCode.java

Newbiekk-kkh committed:
refactor : 로그인 구현에 맞춰 match CRUD 리팩토링

매칭 요청, 매칭 수락/거절, 받은 매칭 조회, 보낸 매칭 조회, 매칭 취소(삭제) 에서 로그인한 유저객체를 받도록 리팩토링

Newbiekk-kkh committed:
refactor : 중복되는 내용의 Dto 들을 MatchResponseDto 로 통합

Newbiekk-kkh committed:
refactor : 팀 코드컨벤션에 맞게 MatchController 수정

MatchController 에 팀 코드컨벤션에 맞지 않는 부분 수정

yeji-world commented on src/main/java/com/example/gamemate/domain/match/controller/MatchController.java:
```diff
@@ -0,0 +1,94 @@
+            @AuthenticationPrincipal CustomUserDetails customUserDetails
+    ) {
+
+        matchService.updateMatch(id, dto,customUserDetails.getUser());
```
dto 뒤에 띄어쓰기 한번 부탁드립니다~

Newbiekk-kkh replied:
수정했습니다

Newbiekk-kkh committed:
feat : HttpMessageNotReadableException 전역 예외처리 추가

Enum 을 Dto 로 받는 매칭 수락/거절에서 발생하는 HttpMessageNotReadableException 을 BadRequest 로 예외처리
